tools/build_release.py

```diff
@@ -21,109 +20,137 @@
+    )
+    parser.add_option(
+        "-j", "--jobs",
+        type="int",
+        dest="jobs",
+        default=1,
+        help="Number of concurrent jobs (default 1)."
+        " Use 0 for auto based on host machine's number of CPUs"
+    )
+    parser.add_option(
+        "-v", "--verbose",
+        action="store_true",
+        dest="verbose",
+        default=False,
+        help="Verbose diagnostic output"
+    )
+    parser.add_option(
+        "-t", "--toolchains",
+        dest="toolchains",
+        help="Use toolchains names separated by comma"
+    )
     parser.add_option("--profile", dest="profile", action="append", default=[])
-
-    parser.add_option("-p", "--platforms", dest="platforms", default="", help="Build only for the platform namesseparated by comma")
-
-    parser.add_option("-L", "--list-config", action="store_true", dest="list_config",
-                      default=False, help="List the platforms and toolchains in the release in JSON")
-
-    parser.add_option("", "--report-build", dest="report_build_file_name", help="Output the build results to an junit xml file")
-
-    parser.add_option("", "--build-tests", dest="build_tests", help="Build all tests in the given directories (relative to /libraries/tests)")
-
-
+    parser.add_option(
+        "-p", "--platforms",
+        dest="platforms",
+        default="",
+        help="Build only for the platform namesseparated by comma"
+    )
+    parser.add_option(
+        "-L", "--list-config",
+        action="store_true",
+        dest="list_config",
+        default=False,
+        help="List the platforms and toolchains in the release in JSON"
+    )
+    parser.add_option(
+        "", "--report-build",
+        dest="report_build_file_name",
+        help="Output the build results to an junit xml file"
+    )
+    parser.add_option(
+        "", "--build-tests",
+        dest="build_tests",
+        help="Build all tests in the given directories"
+        " (relative to /libraries/tests)"
+    )
     options, args = parser.parse_args()
-
-
-
     if options.list_config:
-        print json.dumps(OFFICIAL_MBED_LIBRARY_BUILD, indent=4)
+        print(json.dumps(OFFICIAL_MBED_LIBRARY_BUILD, indent=4))
         sys.exit()
-
     start = time()
     build_report = {}
     build_properties = {}
-
     platforms = None
     if options.platforms != "":
         platforms = set(options.platforms.split(","))
-
     status = True
-
     if options.build_tests:
         # Get all paths
         directories = options.build_tests.split(',')
         for i in range(len(directories)):
             directories[i] = normpath(join(TEST_DIR, directories[i]))
-
         test_names = []
-
-        for test_id in TEST_MAP.keys():
+        for test_id in list(TEST_MAP.keys()):
             # Prevents tests with multiple source dirs from being checked
-            if isinstance( TEST_MAP[test_id].source_dir, basestring):
+            if isinstance(TEST_MAP[test_id].source_dir, str):
```

Comment:
Was this a lint thing? Changing `basestring` to `str`?

Reply:
a 2to3 thing. I'll revert it.

Reply:
I believe we should revert this change to `basestring`
```suggestion
            if isinstance(TEST_MAP[test_id].source_dir, basestring):
```